src/features/pythonApi.ts

```diff
@@ -216,11 +221,75 @@ class PythonEnvironmentApiImpl implements PythonEnvironmentApi {
+    ): Promise<void> {
+        // get the package manager to exit if not found
         const manager = this.envManagers.getPackageManager(context);
         if (!manager) {
             return Promise.reject(new Error('No package manager found'));
         }
+        // what does it mean to return, will we tell the calling extension about it?
+        //check to see if pkg was already installed?
+        const callingExtension = getCallingExtension();
+        traceInfo(`Python API: Installing packages for extension: ${callingExtension}`);
+        const config = getConfiguration('python-envs');
+        let extPkgTrustConfig: SettingsPackageTrust | undefined =
+            config.get<SettingsPackageTrust>('extensionPackageTrust');
+        let callingExtensionTrustLevel;
+        let isConfigured = true;
+        if (extPkgTrustConfig === undefined) {
+            // TODO:s THIS DOESN'T WORK
+            // no package trust config, default to alwaysAsk
+            callingExtensionTrustLevel = 'alwaysAsk';
+            isConfigured = false;
+        } else {
+            // check for package trust settings
+            callingExtensionTrustLevel = extPkgTrustConfig[callingExtension];
+            if (callingExtensionTrustLevel === undefined) {
+                // no specific package trust settings, checking wildcard in config
+                callingExtensionTrustLevel = extPkgTrustConfig['*'];
+                if (callingExtensionTrustLevel === undefined) {
+                    // no wildcard in config, default to alwaysAsk
+                    callingExtensionTrustLevel = 'alwaysAsk';
+                    isConfigured = false;
+                }
+            }
+        }
+        traceInfo(`package trust settings for ${callingExtension} is ${callingExtensionTrustLevel}`);
+
+        if (!isConfigured) {
+            // calling extension has no config, user has no wildcard setup
+            // prompt user to "alwaysAsk" or "alwaysAllow"
+            const selectedOption = await promptForInstallPermissions(callingExtension, packages.join(', '));
+            if (selectedOption === 'cancel') {
+                // user cancelled the prompt, exit
+                window.showErrorMessage(`Package installation of ${packages.join(', ')} was canceled by the user.`);
+                return Promise.resolve();
+            }
+            if (selectedOption !== 'installNoConfigure') {
+                // meaning the user selected "alwaysAsk" or "alwaysAllow", update the config
+                const newExtTrustConfig = { ...extPkgTrustConfig, [callingExtension]: selectedOption };
+                config.update('extensionPackageTrust', newExtTrustConfig, true);
+            }
+        } else {
+            // user has already configured package trust settings for this extension
+            if (callingExtensionTrustLevel === 'alwaysAsk') {
+                traceInfo('Package installation is pending user confirmation due to trust settings.');
```

Comment:
Should we swap out trust here for consistency? 
```suggestion
                traceInfo('Package installation is pending user confirmation due to permission settings.');
```